### PR TITLE
Implement ReadDirAll

### PR DESCRIPTION
**Pull request by mberhault**

### CODE REVIEW

File: filesystem/fs.go [hunk 1]
```diff
@@ -52,17 +53,17 @@ type CFS struct {
 	db *sql.DB
 }
 
-func (fs CFS) initSchema() error {
-	_, err := fs.db.Exec(fsSchema)
+func (cfs CFS) initSchema() error {
```
Comment (mberhault) [inline filesystem/fs.go]: I renamed `fs` to `cfs` everywhere to avoid conflicts with `fuse/fs`

File: filesystem/fs.go [hunk 2]
```diff
@@ -77,51 +78,56 @@ INSERT INTO fs.namespace VALUES ($3, $4, $1);
...
 	if err != nil {
 		return nil, err
 	}
 
-	var results []string
+	var results []fuse.Dirent
 	for rows.Next() {
 		var name string
-		var id int64
+		var id uint64
```
Comment (petermattis) [inline filesystem/fs.go]: You could create the dirent var here and scan directly into it:

```
  dirent := fuse.Dirent{Type: fuse.DT_Unknown}
  if err := rows.Scan(&dirent.Name, &dirent.Inode); err != nil {
    ...
  }
```

File: filesystem/node.go [hunk 3]
```diff
@@ -32,9 +32,12 @@ var _ = fs.Node(&Node{})
 // Lookup
 var _ = fs.NodeStringLookuper(&Node{})
 
+// HandleReadDirAller
+var _ = fs.HandleReadDirAller(&Node{})
```
Comment (petermattis) [inline filesystem/node.go]: We usually write this as: `var _ fs.HandleReaderDirAller = &Node{}`

File: filesystem/node.go [hunk 4]
```diff
@@ -32,9 +32,12 @@ var _ = fs.Node(&Node{})
...
+
 // Node implements the Node interface.
 type Node struct {
```
Comment (petermattis) [inline filesystem/node.go]: We're going to need separate nodes for directories and files.

Comment (petermattis) [inline filesystem/node.go]: I think.

**Comment by petermattis** [comment]
LGTM

File: filesystem/fs.go (hunk 2 above)
Comment (mberhault) [inline filesystem/fs.go]: Done.

File: filesystem/node.go (hunk 3 above)
Comment (mberhault) [inline filesystem/node.go]: Done.

File: filesystem/node.go (hunk 4 above)
Comment (mberhault) [inline filesystem/node.go]: I'm not certain about that. If we use Node to implement both Handler and Node interfaces, we should be fine. still TBD.

**Pull request merged**